src/libraries/System.Memory/tests/Span/IndexOfAny.char.cs

```diff
@@ -772,5 +773,70 @@ public static void MakeSureNoChecksGoOutOfRangeMany_Char()
                 Assert.Equal(-1, index);
             }
         }
+
+        [Fact]
+        public static void TestIndexOfAny_RandomInputs_Char()
+        {
+            IndexOfAnyCharTestHelper.TestRandomInputs(
+                expected: IndexOfAnyReferenceImpl,
+                actual: (searchSpace, values) => searchSpace.IndexOfAny(values));
+
+            static int IndexOfAnyReferenceImpl(ReadOnlySpan<char> searchSpace, ReadOnlySpan<char> values)
+            {
+                for (int i = 0; i < searchSpace.Length; i++)
+                {
+                    if (values.Contains(searchSpace[i]))
+                    {
+                        return i;
+                    }
+                }
+
+                return -1;
+            }
+        }
+    }
+
+    public static class IndexOfAnyCharTestHelper
+    {
+        public delegate int IndexOfAnySearchDelegate(ReadOnlySpan<char> searchSpace, ReadOnlySpan<char> values);
+
+        public static void TestRandomInputs(IndexOfAnySearchDelegate expected, IndexOfAnySearchDelegate actual)
+        {
+            TestRandomInputs(127, expected, actual);
+            TestRandomInputs(char.MaxValue, expected, actual);
+        }
+
+        private static void TestRandomInputs(int maxNeedleValue, IndexOfAnySearchDelegate expected, IndexOfAnySearchDelegate actual)
+        {
+            maxNeedleValue++;
+
+            Span<char> needleSpace = stackalloc char[8];
+            Span<char> haystackSpace = stackalloc char[40];
+            var rng = new Random();
+
+            for (int iterations = 0; iterations < 100_000; iterations++)
```

Comment:
How long do these tests take to run?  Wondering if they meaningfully increase execution time of the test suite and should be made outerloop.